reason_4.0/lib/core/content_managers/thor.php3

```diff
@@ -178,6 +178,30 @@
 			{
 				$publish_status_text .= '<strong>Status:</strong> Unpublished. <a target="_blank" href="http://reasoncms.org/userdocs/managing-content/other-types/forms/#attaching_a_form_to_a_page">How to publish your form</a>';
 			}
+
+			$form_entity = new entity ($this->admin_page->id);
+			$thor_view_value = $form_entity->get_value('thor_view');
+
+			if ($this->db_table_exists_check()) {
+				if (empty($thor_view_value)) {
+					$form_entity = new entity ($this->admin_page->id);
+					$old_thor_content = $form_entity->get_value('thor_content');
+					$new_thor_content = ($this->get_value('thor_content')) ? $this->get_value('thor_content') : $old_thor_content;
+					if ($new_thor_content != $old_thor_content) {
+						$data_manager_link = unhtmlentities($this->admin_page->make_link(array('cur_module' => 'ThorData')));
+						$this->set_error('thor_content', 'Changes could not be saved because of associated data. You can change the form contents if you first <a href="' . $data_manager_link . '">delete the data</a> associated with the form.');
+						$this->show_error_jumps = false;
+					}
+					$data_manager_link = $this->admin_page->make_link(array('cur_module' => 'ThorData'));
+					$publish_status_text .= '<p><strong>This form has stored data. </strong><a href="' . $data_manager_link . '">Manage stored data</a><br>You can add new form fields and modify existing fields, but if you remove a field, the removed field\'s data will persist until you delete the stored data for this form. Changing the type of an existing field is unsupported.</p>';
```

Comment:
I don't think we need to specify what's the same as when you don't have stored data, just what's different. Suggested text edit:

If you remove a field, the field's data will remain available until you delete the stored data for this form. Changing existing fields' types is not supported.

Reply:
I just saw Doug's suggestion, which is better. So the message would be:

If you remove a field, the field's data will remain available until you delete the stored data for this form. You cannot change the type of existing fields.